specutils/manipulation/resample.py

```diff
@@ -73,59 +74,151 @@ class FluxConservingResampler(ResamplerBase):
     >>> fluxc_resample(input_spectra, resample_grid)  # doctest: +FLOAT_CMP
-    <Spectrum1D(flux=<Quantity [        nan,  3.        ,  6.13043478,  7.        ,  6.33333333,
-               10.        , 20.        ,         nan,         nan] mJy>, spectral_axis=<SpectralAxis [ 1.,  5.,  9., 13., 14., 17., 21., 22., 23.] nm>)>
-
+    <Spectrum1D(flux=<Quantity [  nan,  3.  ,  6.  ,  7.  ,  6.25, 10.  , 20.  ,   nan,   nan] mJy>, spectral_axis=<SpectralAxis [ 1.,  5.,  9., 13., 14., 17., 21., 22., 23.] nm>)>
     """
 
-    def _resample_matrix(self, orig_spec_axis, fin_spec_axis):
+    def _fluxc_resample(self, input_bin_centers, output_bin_centers,
+                        input_bin_fluxes, errs):
         """
-        Create a re-sampling matrix to be used in re-sampling spectra in a way
-        that conserves flux. This code was heavily influenced by Nick Earl's
-        resample rough draft: nmearl@0ff6ef1.
+        Resample ``input_bin_fluxes`` and (optionally) ``errs`` from
+        ``input_bin_centers`` to ``output_bin_centers``.
 
         Parameters
         ----------
-        orig_spec_axis : SpectralAxis
-            The original spectral axis array.
-        fin_spec_axis : SpectralAxis
-            The desired spectral axis array.
+        input_bin_centers : `~specutils.SpectralAxis`
+            `~specutils.SpectralAxis` object, with input bin centers.
+        output_bin_centers : `~specutils.SpectralAxis`
+            `~specutils.SpectralAxis` object, with input bin centers.
+        input_bin_fluxes : Quantity
+            Quantity array of flux values.
+        errs : `~astropy.nddata.Variance` object, or None
+            Variance array of errors corresponding to input bin fluxes. If None,
+            error resampling is not performed.
 
         Returns
         -------
-        resample_mat : ndarray
-            An [[N_{fin_spec_axis}, M_{orig_spec_axis}]] matrix.
+       (output_fluxes, output_errs)
+            A tuple containing plain numpy arrays of the resampled fluxes and
+            errors (if available).
         """
-        # Lower bin and upper bin edges
-        orig_edges = orig_spec_axis.bin_edges
-        fin_edges = fin_spec_axis.bin_edges
 
-        # I could get rid of these alias variables,
-        # but it does add readability
-        orig_low = orig_edges[:-1]
-        fin_low = fin_edges[:-1]
-        orig_upp = orig_edges[1:]
-        fin_upp = fin_edges[1:]
+        fill_val = np.nan  # bin_edges=nan_fill case
+        if self.extrapolation_treatment == 'zero_fill':
+            fill_val = 0
+
+        # get bin edges from centers
+        input_bin_edges = input_bin_centers.bin_edges.value
+        output_bin_edges = output_bin_centers.bin_edges.value
+
+        # convert to regular arrays
+        input_bin_centers = input_bin_centers.value
+        output_bin_centers = output_bin_centers.value
+        input_bin_fluxes = input_bin_fluxes.value
```

Comment:
Is it actually necessary to use `.value` here? It doesn't matter too much given the conversion to match units has already happened, but if it doesn't have a negative consequence I think it's safer not to use `.value` since there's always the chance that this will catch some unit conversion issues that slip through.

Reply:
shouldn't be necessary, i was getting a bunch of TypeErrors i didn't want to track down haha. i'll look into it

Reply:
Done!